app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    dict__ = dict()
+
+    @wraps(func)
+    def inner(*args) -> Any:
+        nonlocal dict__
```

Comment:
As noted below, using `args` instead of `(*args, inner.__name__,)` as the key in the dictionary would simplify the code and make it easier to understand, and this would render the use of `nonlocal` unnecessary.

Also, Using `nonlocal`  or `global` statements to modify variables declared in an enclosing scope can lead to difficulties in understanding and maintaining the code, and can also make it harder to test and debug. You should use 'em if you really need to.

Therefore, it's generally better to avoid using nonlocal or global statements whenever possible and to use other techniques